### ISSUE TITLE
`SkippedByTrustPolicy` from JSON output of `notation verify` is not documented

### ISSUE THREAD

**Issue by shizhMSFT** [notaryproject/notation#547]
### What is the areas you experience the issue in?

Notation CLI

### What is not working as expected?

The result status `SkippedByTrustPolicy` from the JSON output of `notation verify` is not documented.

Although the follow-up scripts or program can determine the verification status by the return code, developers do not know there is a value `SkippedByTrustPolicy` besides `Success` unless they read through the source code.

### What did you expect to happen?

Have the JSON output of `notation verify` documented, including the enums of the result status.

**Comment by byronchien** [comment]
if a "reason" field is being added for [this issue](https://github.com/notaryproject/notation/issues/546), then instead of having "SkippedByTrustPolicy" as the result, will just use "Skipped" and mention the trust policy in the reason. Will still add documentation for the different result values.

**Comment by sajayantony** [comment]
Can we also improve the output of -v 

I think we can improve the user experience by highliging key areas that are important and removing unnecessary golang masrhalling info. /cc @shizhMSFT 

``` 
signer@3344ae675ce7:/notation-demos$ notation verify -v $IMAGE
INFO Using the referrers tag schema
INFO Reference sha256:934e37d5c1029186129234a1af1d46b90a9d6b14c02276536e6e9b2ec98db3a9 resolved to manifest descriptor: {MediaType:application/vnd.oci.image.manifest.v1+json Digest:sha256:934e37d5c1029186129234a1af1d46b90a9d6b14c02276536e6e9b2ec98db3a9 Size:507 URLs:[] Annotations:map[] Data:[] Platform:<nil> ArtifactType:}
INFO Checking whether signature verification should be skipped or not
INFO Trust policy configuration: &{Name:notation-demos-images RegistryScopes:[localhost:5000/hello] SignatureVerification:{VerificationLevel:strict Override:map[]} TrustStores:[ca:notation-demos.io] TrustedIdentities:[*]}
INFO Check over. Trust policy is not configured to skip signature verification
INFO Processing signature with manifest mediaType: application/vnd.oci.image.manifest.v1+json and digest: sha256:8d11c5ea366ac877d2a8fd8505a2445a406710cddb91504f0156238ceff45195
INFO Trust policy configuration: &{Name:notation-demos-images RegistryScopes:[localhost:5000/hello] SignatureVerification:{VerificationLevel:strict Override:map[]} TrustStores:[ca:notation-demos.io] TrustedIdentities:[*]}
Successfully verified signature for localhost:5000/hello@sha256:934e37d5c1029186129234a1af1d46b90a9d6b14c02276536e6e9b2ec98db3a9
```

**Comment by yizha1** [comment]
Closed by #551
